### PR TITLE
Add a backoff retry mechanism for starting runners

### PR DESCRIPTION
**Pull request by ffilippopoulos**

### CODE REVIEW

File: backoff/backoff.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
+// Package sidecar includes this backoff function copied from: https://github.com/jpillora/backoff/blob/d80867952dff4e2fbfb4280ded4ff94d67790457/backoff.go
```
Comment (ribbybibby) [inline backoff/backoff.go:1]:
```suggestion
// Package main includes this backoff function copied from: https://github.com/jpillora/backoff/blob/d80867952dff4e2fbfb4280ded4ff94d67790457/backoff.go
```

Comment (ffilippopoulos) [reply backoff/backoff.go:1]: hah, so obvious I copied from vault-sidecar there

File: backoff/retry.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+type Operation func() error
+
+const (
+	defaultBackoffJitter = true
+	defaultBackoffMin    = 2 * time.Second
+	defaultBackoffMax    = 1 * time.Minute
+)
+
+func RetryWithDefaultBackoff(op Operation, description string) {
```
Comment (ribbybibby) [inline backoff/retry.go:17]:
```suggestion
func Retry(op Operation, description string) {
```
Nitpick. I don't think you need to be explicit about the fact that it uses defaults.

Comment (ffilippopoulos) [reply backoff/retry.go:17]: hmm I like the explicit names, but I do not have a strong argument. In that case I guess we should only care to export `Retry` from the package and hide everything else.

Comment (ribbybibby) [reply backoff/retry.go:17]: I think exporting `backoff.Retry` and `backoff.RetryWithBackoff` is fine.

My thinking is that, i.e `http.Get` doesn't need to be called `http.GetWithDefaultClientAndSettings` for me to understand that it's going to use some defaults.

**Review by ribbybibby — APPROVED** [review]
lgtm

**Pull request merged**